### DISCUSSION TITLE
Dependencies

### DISCUSSION THREAD

**Discussion by Otoru** [vitalik/django-ninja#129] [Q&A] [answered]
What are django-ninja's dependencies on `INSTALLED_APPS`?

**Comment by vitalik** [comment, marked as answer]
@Otoru 

You do not have to put ninja to `INSTALLED_APPS`

in this  case swagger docs javascirpt and css will be loaded from CDN

but if you put `ninja` to `INSTALLED_APPS` - js/css will be loaded from local staticfiles backed

**Reply by Otoru** [reply]
Since he doesn't need to be there, I'm questioning what dependencies django-ninja needs in `INSTALLED_APPS`. For example `django.contrib.staticfiles` or `django.contrib.messages`.

Is there no dependency?

**Reply by vitalik** [reply]
I guess the only dependency is Session + Authentication middlewares if you are going to use `ninja.security.django_auth`
other middlewares are not required

django inja is mainly bulit on top of url-resolver + pydantic

**Reply by Otoru** [reply]
I'm going to do some more testing with this, but if that's the case, the developers are to be congratulated.
This lack of strings is wonderful <3
